### PR TITLE
techdebt: [CI-8602]: Remove KubernetesHosted Infra code path

### PR DESCRIPTION
**Pull request by sahithibanda01**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51331)

### CODE REVIEW

**Review by hemanthmantri — COMMENTED** [review]
Didn't we deprecate k8s hosted infra? Also, I think you have changes in ci-manager config.yml which are supposed to be local.

**Comment by sahithibanda01** [comment]
> Didn't we deprecate k8s hosted infra? Also, I think you have changes in ci-manager config.yml which are supposed to be local.

We are removing the code path for K8Hosted Infra and also will remove my local changes in this PR

File: 120-ng-manager/config.yml [hunk 1]
```diff
@@ -549,5 +549,4 @@ webhookBranchHookEventHsqsDequeueConfig:
   batchSize: ${WEBHOOK_BRANCH_HOOK_EVENT_DEQUE_BATCH_SIZE:-20}
   threadSleepTimeInMillis: ${WEBHOOK_BRANCH_HOOK_EVENT_DEQUE_THREAD_SLEEP_TIME_IN_MILLIS:-30}
 
-useQueueServiceForWebhookTriggers: ${USE_QUEUE_SERVICE_FOR_WEBHOOK_TRIGGERS:-false}
-
+useQueueServiceForWebhookTriggers: ${USE_QUEUE_SERVICE_FOR_WEBHOOK_TRIGGERS:-false}
```
Comment (raghavharness) [inline 120-ng-manager/config.yml:552]: you can undo this change

**Comment by sahithibanda01** [comment]
trigger uts

**Comment by sahithibanda01** [comment]
trigger ut0
trigger ut1
trigger ut5

**Comment by sahithibanda01** [comment]
trigger ut0,ut1,ut5

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/execution/CIExecutionConfigService.java [hunk 2]
```diff
@@ -385,17 +385,19 @@ private CIExecutionImages mapK8Config(CIExecutionConfig config) {
         .build();
   }
 
-  public List<DeprecatedImageInfo> getDeprecatedTags(String accountId) {
```
Comment (raghavharness) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/execution/CIExecutionConfigService.java:388]: Revert this

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/integrationstage/K8InitializeStepUtils.java [hunk 3]
```diff
@@ -428,8 +427,7 @@ public ContainerDefinitionInfo createPluginCompatibleStepContainerDefinition(Plu
   private void setEnvVariablesForHostedBuids(
       IntegrationStageNode stageNode, PluginCompatibleStep stepInfo, Map<String, String> envVarMap) {
     IntegrationStageConfig stage = stageNode.getIntegrationStageConfig();
-    if (stage != null && stage.getInfrastructure() != null
-        && stage.getInfrastructure().getType() == Infrastructure.Type.KUBERNETES_HOSTED) {
+    if (stage != null && stage.getInfrastructure() != null) {
```
Comment (raghavharness) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/integrationstage/K8InitializeStepUtils.java:430]: Remove the if condition

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/integrationstage/K8InitializeStepUtils.java [hunk 4]
```diff
@@ -446,8 +444,7 @@ private void setEnvVariablesForHostedBuids(
   private void setEnvVariablesForHostedCachingSteps(
       IntegrationStageNode stageNode, String identifier, Map<String, String> envVarMap) {
     IntegrationStageConfig stage = stageNode.getIntegrationStageConfig();
-    if (stage != null && stage.getInfrastructure() != null
-        && stage.getInfrastructure().getType() == Infrastructure.Type.KUBERNETES_HOSTED) {
+    if (stage != null && stage.getInfrastructure() != null) {
```
Comment (raghavharness) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/integrationstage/K8InitializeStepUtils.java:447]: ditto

**Review by raghavharness — CHANGES_REQUESTED** [review]

**Review by raghavharness — APPROVED** [review]

**Comment by raghavharness** [comment]
trigger uts

**Comment by sahithibanda01** [comment]
trigger uts

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/buildstate/BuildSetupUtils.java [hunk 5]
```diff
@@ -36,8 +36,6 @@ public CIInitializeTaskParams getBuildSetupTaskParams(
       InitializeStepInfo initializeStepInfo, Ambiance ambiance, String logPrefix) {
     switch (initializeStepInfo.getInfrastructure().getType()) {
       case KUBERNETES_DIRECT:
-      case KUBERNETES_HOSTED:
-        return k8InitializeTaskParamsBuilder.getK8InitializeTaskParams(initializeStepInfo, ambiance, logPrefix);
```
Comment (raghavharness) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/buildstate/BuildSetupUtils.java:40]: You should not be removing the return statement from here

Comment (raghavharness) [reply 332-ci-manager/service/src/main/java/io/harness/ci/execution/buildstate/BuildSetupUtils.java:40]: Let's also run a k8 pipeline on local to verify the flow once.

**Comment by sahithibanda01** [comment]
trigger uts

**Review by raghavharness — APPROVED** [review]

**Comment by harshjain12** [comment]
@raghavharness Shall we test hosted vm once before merging it ?

**Comment by raghavharness** [comment]
> @raghavharness Shall we test hosted vm once before merging it ?

@harshjain12 Has now tested for both k8 and VM. Can you review once.

**Pull request merged**